bundles/remote_services/remote_service_admin_shm_v2/rsa_shm/src/rsa_shm_client.c

```diff
@@ -179,8 +181,8 @@ void rsaShmClientManager_destory(rsa_shm_client_manager_t *clientManager) {
     size_t listSize = celix_arrayList_size(clientManager->exceptionMsgList);
     for (int i = 0; i < listSize; ++i) {
         rsa_shm_exception_msg_t *exceptionMsg = celix_arrayList_get(clientManager->exceptionMsgList, i);
-        shmPool_free(clientManager->shmPool, exceptionMsg->msgBuffer);
-        rsaShmClientManager_destroyMsgControl(clientManager, exceptionMsg->msgCtrl);
+        //Here，we should not free 'msgBuffer' and 'msgCtrl' by shmPool_free, because rsa_shm_server maybe using them, and tlsf_free will modify freed memory.
+        //The shared memory  of 'msgBuffer' and 'msgCtrl' will be freed automatically when nobody is using it.
```

Comment:
I consider it a design flaw, i.e., lack of synchronization between server and client.

Of course, if `pthread_cond_destroy` and `pthread_mutex_destroy` does not release any resource like kernel objects, then no real issue should arise.